### DISCUSSION TITLE
FFT on audio data - force data length?

### DISCUSSION THREAD

**Discussion by Grassbaecker** [pschatzmann/ESP32-A2DP#219] [Q&A] [answered]
is it posible to force the data length that the A2DP source will send my Sink
couse my mobil send me 3800 and some bytes and wish it would send me smaller packets and so my sample colector would not colect so much heap to reach to reach the wished samples

with regarts bernhard

**Comment by pschatzmann** [comment]
No, that is all decided by A2DP and you don't have any means to influence this. 
Why would you collect any data ? What are you planning to do ?
You would usually process the data right away from the buffer that is passed to you...

**Reply by Grassbaecker** [reply]
i colect the data for the FFT but i wish to analyze 1024 samples    but the phone sent me about 980 samples per packet and so m< framerate go under 20 FPS while the when the data come from analogin wher the buffer is 1024 byte and i reach afer 4 packts my wished samples i come to 33 FPS
but ok i will set my FFT size to 512 iN A2DP mode where i reach with every packt my samples but the bass resolution is suffering

thanks for info Bernhard

**Comment by pschatzmann** [comment]
If you use the FFT functionality of my AudioTools library, this should get very simple to implement. You just write what you have. 
There are no memory savings though...
https://github.com/pschatzmann/arduino-audio-tools/wiki/FFT
https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit/streams-audiokit-fft

**Reply by Grassbaecker** [reply]
as i tolt you i can not use your AudioFFT couse proplem with my oled
Streamcopy is absolut not usable for me couse only one destination i can not push samples to FFT and also to output with.
and my trys today with RealFFT look not good couse im to stupid for this lib and dont know how to start

thanks in advance Bernhard

**Comment by pschatzmann** [comment, marked as answer]
Strange, I don't see any mention of oled in this issue. 

There is no limitation about the number of outputs.
- If you use th callback approach like https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-basic-api/basic-a2dp-i2s there is just one more destination to write to.
- If you use the stream based approach like in https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-a2dp-serial you can just use a [MultiOutput](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_multi_output.html) which defines all final outputs (as many as you want)

The first approach uses less memory.

If you tell me what's unclear about the FFT documentation or example I might be able to improve it...
